src/services/map.js

```diff
@@ -1,12 +1,14 @@
 import { createWorldTerrain, Viewer } from 'cesium';
+import AddMarkers from './add-markers';
 
 export default function map() {
```

Comment:
As I am using useStore from Zustand inside this function, I had to disable the react-hooks/rules-of-hooks to change the component AddMarkers to a function addMarkers - don't know, if this is what you meant, though... ?
-> Nevertheless it is done in commit 'Correct marker issues'

The DOM element I am searching by className is not created by me, but by cesium...therefore I don't know, how to give it another attribute. I needed to detect if this element is already there to not having two globes showing up. 
And the addMarkers-funktion needs the viewer-name as input to relate the entities to it - I have no idea, how this can be done when this function is in the Map-component. Not even regarding, that everything fired by useEffect in the Map-component is fired double.
In the end I changed everything back again and I really hope, you can teach me a better solution as I am not happy with it either, but I cannot find out how to change it...

Reply:
Very important: This code should be a component, not a service! 

Logic can be wrapped inside a hook to use other hooks. (see my comment in `add-markers.js`)

You should create a ref for the  Element `id=cesiumContainer` Use the ref as root for `querySelector`. This logic should be wrapped inside an effect. More ore less like this:

```
// component logic
const cesiumContainerRef = useRef();

useEffect(()=>{
	if(cesiumContainerRef){
		 const childNo = cesiumContainerRef.queySelector(‘.cesium-viewer’);
		// ...
	}
} [cesiumContainerRef])


// rendering
<MapContainer ref={cesiumContainerRef} id="cesiumContainer" />
```

Reply:
Done in commit 'Correct Map issues'